vcd/resource_vcd_service_account.go

```diff
@@ -0,0 +1,344 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdServiceAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdServiceAccountCreate,
+		UpdateContext: resourceVcdServiceAccountUpdate,
+		ReadContext:   resourceVcdServiceAccountRead,
+		DeleteContext: resourceVcdServiceAccountDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdServiceAccountImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of service account",
+			},
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"software_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Any valid UUID, depends on the user, e.g: 12345678-1234-5678-90ab-1234567890ab",
+			},
+			"role_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Role ID of service account",
+			},
+			"software_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Version of software using the service account, can be freely defined by the user",
+			},
+			"uri": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "URI of the client using the service account, can be freely defined by the user",
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Status of the service account.",
+			},
+			"file_name": {
```

Comment:
Should it have a `ForceNew: true` flag? If we change filename, we're not effectively renaming the token filename.

Reply:
Makes sense, added.